processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -97,19 +99,23 @@
       return String.class;
     }
 
+    /**
+     * numBytes will be {@link NULL_VALUE_SIZE_MARKER} for null values.
+     */
     @Override
+    @Nullable
     public String fromByteBuffer(final ByteBuffer buffer, final int numBytes)
     {
-      return StringUtils.fromUtf8(buffer, numBytes);
+      if (numBytes < 0) {
```

Comment:
It's hard to comprehend the logic of this method. Let's clarify concerns. There are two possible formats: legacy and the new, nullable, and two possible configurations: sqlCompatible and replaceWithDefault. So 4 options. Could you please write what happens in all 4 of them for two cases: numBytes < 0 and numBytes = 0. (in a comment in this method)

Reply:
added comment.